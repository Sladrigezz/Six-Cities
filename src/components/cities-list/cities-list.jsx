import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {ActionCreator} from '../../reducers/index.js';
import {updateOffersList as updateOffers} from '../../reducers/data.js';

export const CitiesList = ({cities, activeCity, setActiveCity, updateOffersList}) => {
  const onItemClickHandler = (e) => {
    const cityName = e.currentTarget.id;
    setActiveCity(cityName);
    updateOffersList();
  };

  return (
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {cities.map((item) => {
          const isActive = activeCity === item;
          return (
            <li className="locations__item" key={item} >
              <a
                className={`locations__item-link tabs__item ${isActive ? `tabs__item--active` : ``}`}
                id={item}
                onClick={onItemClickHandler}
              >
                <span>{item}</span>
              </a>
            </li>
          );
        })}
      </ul>
    </section>
  );
}


CitiesList.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string),
  activeCity: PropTypes.string,
  updateOffersList: PropTypes.func,
  setActiveCity: PropTypes.func,
};

export default connect(
  (state) => ({
    activeCity: state.user.activeCity,
    cities: state.user.citiesList,
    state,
  }),
  (dispatch) => ({
    setActiveCity: (cityName) => dispatch(ActionCreator.setCity(cityName)),
    updateOffersList: () => dispatch(updateOffers()),
  })
)(CitiesList);
