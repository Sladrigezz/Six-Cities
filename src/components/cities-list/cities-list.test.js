import React from 'react';
import renderer from 'react-test-renderer';
import {CitiesList} from './cities-list.jsx';

it(`should match snapshot`, () => {
  const setActiveCity = jest.fn();
  const updateOffersList = jest.fn();
  const props = {
    cities: [`Paris`, `AAA`],
    activeCity: `AAA`,
    setActiveCity,
    updateOffersList,
  };

  const wrapper = renderer.create(<CitiesList {...props} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
