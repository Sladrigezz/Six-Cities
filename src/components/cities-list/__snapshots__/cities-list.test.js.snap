// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        id="Paris"
        onClick={[Function]}
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        id="AAA"
        onClick={[Function]}
      >
        <span>
          AAA
        </span>
      </a>
    </li>
  </ul>
</section>
`;
